sql-plugin/src/main/scala/com/nvidia/spark/rapids/window/GpuWindowExpression.scala

```diff
@@ -1468,8 +1478,16 @@ class SumBinaryFixer(toType: DataType, isAnsi: Boolean)
             }
           }
           withResource(nullsReplaced) { nullsReplaced =>
-            withResource(nullsReplaced.binaryOp(BinaryOp.ADD, prev, prev.getType)) { updated =>
-              mask.ifElse(updated, windowedColumnOutput)
+            if (needsBasicOverflowCheck) {
+              withResource(nullsReplaced.binaryOp(BinaryOp.ADD, prev, prev.getType)) { updated =>
+                val ret = mask.ifElse(updated, windowedColumnOutput)
+                AddOverflowChecks.basicOpOverflowCheck(updated, prev, ret, Some(mask))
```

Comment:
A possible column leak.
```suggestion
                closeOnExcept(ret)(_ => AddOverflowChecks.basicOpOverflowCheck(updated, prev, ret, Some(mask)))
```